File: app/src/main/java/openfoodfacts/github/scrachx/openfood/dagger/module/FragmentModule.java
package org.openfoodfacts.scanner.dagger.module;

import dagger.Module;
import dagger.Provides;
import org.openfoodfacts.scanner.dagger.FragmentScope;
import org.openfoodfacts.scanner.views.viewmodel.category.CategoryFragmentViewModel;

@Module
public class FragmentModule {
    @FragmentScope
    @Provides
    CategoryFragmentViewModel provideCategoryFragmentViewModel() {
        return new CategoryFragmentViewModel();
    }
}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/dagger/module/FragmentModule.kt
package openfoodfacts.github.scrachx.openfood.dagger.module

import dagger.Module
import dagger.Provides
import openfoodfacts.github.scrachx.openfood.dagger.FragmentScope
import openfoodfacts.github.scrachx.openfood.features.viewmodel.category.CategoryFragmentViewModel

@Module
class FragmentModule {
    @FragmentScope
    @Provides
    fun provideCategoryFragmentViewModel(): CategoryFragmentViewModel = CategoryFragmentViewModel()
}